src/gsy_e/models/market/two_sided.py

```diff
@@ -337,48 +330,59 @@ def match_recommendations(
-            bid: Bid, offer: Offer, clearing_rate: float = None,
-            selected_energy: float = None) -> None:
+            recommendation: BidOfferMatch) -> None:
         """Validate if both trade parties satisfy each other's requirements.
 
         :raises:
             InvalidBidOfferPairException: Bid offer pair failed the validation
         """
-        if ((offer.requirements or bid.requirements) and
-                not RequirementsSatisfiedChecker.is_satisfied(
-                    offer=offer, bid=bid, clearing_rate=clearing_rate,
-                    selected_energy=selected_energy)):
-            # If no requirement dict is satisfied
+        requirements_satisfied = True
+        if (recommendation.matching_requirements or {}).get("offer_requirement"):
+            offer_requirement = recommendation.matching_requirements["offer_requirement"]
+            requirements_satisfied &= RequirementsSatisfiedChecker.is_offer_requirement_satisfied(
+                recommendation.offer, recommendation.bid, offer_requirement,
+                recommendation.trade_rate, recommendation.selected_energy)
+        if (recommendation.matching_requirements or {}).get("bid_requirement"):
+            bid_requirement = recommendation.matching_requirements["bid_requirement"]
+            requirements_satisfied &= RequirementsSatisfiedChecker.is_bid_requirement_satisfied(
+                    recommendation.offer, recommendation.bid, bid_requirement,
+                    recommendation.trade_rate, recommendation.selected_energy)
+        if not requirements_satisfied:
+            # If requirements are not satisfied
             raise InvalidBidOfferPairException(
                 "The requirements failed the validation.")
 
-    @classmethod
     def validate_bid_offer_match(
-            cls, bid: Bid, offer: Offer,
-            clearing_rate: float, selected_energy: float) -> None:
+            self, recommendation: BidOfferMatch) -> None:
         """Basic validation function for a bid against an offer.
 
         Raises:
             InvalidBidOfferPairException: Bid offer pair failed the validation
         """
-        bid_energy = bid.energy
-        offer_energy = offer.energy
+        selected_energy = recommendation.selected_energy
+        clearing_rate = recommendation.trade_rate
+        market_offer = self.offers.get(recommendation.offer["id"])
+        market_bid = self.bids.get(recommendation.bid["id"])
+
+        if not (market_offer and market_bid):
+            # If not all offers bids exist in the market, skip the current recommendation
+            raise InvalidBidOfferPairException("Not all bids and offers exist in the market.")
+        bid_energy = recommendation.bid_energy
+        offer_energy = market_offer.energy
         if selected_energy > bid_energy:
             raise InvalidBidOfferPairException(
                 f"Energy traded {selected_energy} is higher than bids energy {bid_energy}.")
```

Comment:
Why are we returning the `bid_energy` here and the `offer_energy` a few lines below in the bid_energy_rate but don't do it in the error messages for the rates?